### DISCUSSION TITLE
Can't get thinkfan to run, doesn't find /sys/devices/pci.../hwmon

### DISCUSSION THREAD

**Discussion by Chrissi43** [vmatare/thinkfan#208] [Support]
Hallo,

I have a huge problem with thinkfan. It's really a great software. So far i have had no problems. Now after I upgraded to a newer linux system I can no longer get the software to run. I'm not a software engineer, only a boring normal computer user. I have no idea what the problem is. Maybe someone here knows a solution?

### My system consists of

- Kernel: 5.10.0-18-amd64 [5.10.140-1] x86_64
- init=/lib/systemd/systemd
- Distro: MX-21.2.1_x64 Wildflower April 9  2022
- base: Debian GNU/Linux 11 (bullseye)
- Machine: Laptop System: LENOVO product: 3680HSG v: ThinkPad X201

### What I have done so far

- I followed the instructions for installation and setup as indicated here: [https://thinkwiki.de/Thinkfan](url) -- I used this instruction for my older system before and everything was fine.
- So I installed the packages "thinkfan" (1.2.1-3~mx21+1) and "lm-sensors" (1:3.6.0-7) from the normal package sources without an error.
- I created the file `/etc/modprobe.d/thinkfan.conf` with the content `options thinkpad_acpi fan_control=1`
- I looked after the temperature control devices with `find /sys/devices -type f -name "temp*_input"`
- Then I created the file `/etc/thinkfan.conf`with these informations. Here is the content of this file:

```
hwmon /sys/devices/platform/thinkpad_hwmon/hwmon/hwmon2/temp6_input
hwmon /sys/devices/platform/thinkpad_hwmon/hwmon/hwmon2/temp3_input
hwmon /sys/devices/platform/thinkpad_hwmon/hwmon/hwmon2/temp7_input
hwmon /sys/devices/platform/thinkpad_hwmon/hwmon/hwmon2/temp4_input
hwmon /sys/devices/platform/thinkpad_hwmon/hwmon/hwmon2/temp8_input
hwmon /sys/devices/platform/thinkpad_hwmon/hwmon/hwmon2/temp1_input
hwmon /sys/devices/platform/thinkpad_hwmon/hwmon/hwmon2/temp5_input
hwmon /sys/devices/platform/thinkpad_hwmon/hwmon/hwmon2/temp2_input
hwmon /sys/devices/platform/coretemp.0/hwmon/hwmon3/temp4_input
hwmon /sys/devices/platform/coretemp.0/hwmon/hwmon3/temp2_input

(0,	0,	60)
(1,	60,	65)
(2,	65,	70)
(3,	70,	75)
(4,	75,	80)
(5,	80,	85)
(7,	85,	32767)
```
- I rebooted my system

### Testing the installation

- After reboot I tested thinkfan with the command `sudo thinkfan -n`
- It gave me this error: `ERROR: Error scanning /sys/devices/pci0000:00/0000:00:03.1/0000:27:00.0/hwmon: No such file or directory`

### Solution?

I am quite perplexed, since everything has run without problems so far with the old system. I also do not find my problem described on the Internet or here in the forum. I have not yet dared to compile the program with the newest version myself. I have no idea about compiling and have not needed it yet. Does anyone know advice?

Thanks in advance for any help,
Chrissi

**Comment by 30m3e** [comment]
Does the thinkpad_acpi load in /etc/modprobe.d properly?

**Comment by Chrissi43** [comment]
Hallo ThaungHan,

thank you for your time and question. I checked with `lsmod` whether "thinkpad_acpi" is loading. Among other things, the output lists the module, but with a zero. So if I understand it right the module is being loaded but not used?

**Comment by 30m3e** [comment]
> Hallo ThaungHan,
> 
> thank you for your time and question. I checked with `lsmod` whether "thinkpad_acpi" is loading. Among other things, the output lists the module, but with a zero. So if I understand it right the module is being loaded but not used?

" I created the file /etc/modprobe.d/thinkfan.conf with the content options thinkpad_acpi fan_control=1 "

I think the thinkpad_acpi.conf path should be in /etc/modprobe.d/ thinkpad_acpi.conf  and  fan config file should be in /etc/thinkfan.conf imo.

And also you should cat /proc/acpi/ibm/fan is enabled or not. If the output is enabled and thing still does not work, you should unload the module first "modprobe -r thinkpad_acpi"and load again "modprobe thinkpad_acpi. I think it would work. Pls keep in mind that I'm also a user and thinkfan is broken many times by my messing up.  Hope it work out for you. Thank you.

**Comment by Chrissi43** [comment]
Thank you for your kindly suggestions.

> I think the `thinkpad_acpi.conf` path should be in `/etc/modprobe.d/ thinkpad_acpi.conf` and fan config file should be in `/etc/thinkfan.conf` imo.

I have also noticed that the naming of the files in the instructions sometimes diverges.  However, so far I had stuck to the old variant that worked for me. I have now as you recommended

1. put the renamed file `thinkpad_acpi.conf` into the directory `/etc/modprobe.d` and
2. the file `thinkfan.conf` into the directory `/etc`.

> And also you should `cat /proc/acpi/ibm/fan` is enabled or not.

- I have run the command, the status is enabled, level auto.
- After that, I restarted the notebook again to be on the safe side.
- After running `lsmod`, output still for `thinkpad_acpi` is a zero.
- The command `sudo thinkfan -n` unfortunately still produces the known error message: `ERROR: Error scanning /sys/devices/pci0000:00/0000:00:03.1/0000:27:00.0/hwmon: No such file or directory`

> If the output is enabled and thing still does not work, you should unload the module first `modprobe -r thinkpad_acpi` and load again `modprobe thinkpad_acpi`.

- I have now unloaded the module and reloaded it as suggested.
- After that a new test with `lsmod` - unchanged display, `thinkpad_acpi` is listed with a zero
- After that a new test with `sudo thinkfan -n` - known error message.

Then I tried something else again. Let me know the status with command `service thinkfan status`. Output:
```
thinkfan.service
Loaded: loaded (/lib/systemd/system/thinkfan.service; disabled; vendor preset: enabled)
Active: failed (Result: exit-code) since Thu 2022-10-13 21:34:51 CEST; 47s ago
Process: 4069 ExecStart=/usr/sbin/thinkfan $THINKFAN_ARGS $DAEMON_ARGS (code=exited, status=1/FAILURE)
CPU: 7ms
```
So the service is loaded but disabled. Could it possibly be because my Linux system runs with systemd?

> I think it would work. Pls keep in mind that I'm also a user and thinkfan is broken many times by my messing up. Hope it work out for you. Thank you.

In the end, I think it doesn't matter whether you are a programmer or "just" a user - what counts in the end is that a solution is found? In this sense, thank you very much for your help, "four eyes see more than two". :-)

**Comment by Chrissi43** [comment]
- So, have now once booted my Linux system without `systemd`.
- The command `lsmod` lists unchanged `thinkpad_acpi` with a zero.
- The command `sudo thinkfan -n` gives the well-known error like before: `ERROR: Error scanning /sys/devices/pci0000:00/0000:00:03.1/0000:27:00.0/hwmon: No such file or directory`
- The command `sudo service thinkfan start` gives the same error.
- The command `service thinkfan status` gives the output: `thinkfan is not running ... failed!`

So no progress. :-(

**Comment by vmatare** [comment]
@Chrissi43 have you taken a look at your hwmon hierarchy to see what's actually in there? I'd recommend looking at `/sys/class/hwmon`. Either use those paths directly or look where the symlinks point.

**Comment by vmatare** [comment]
Oh and please don't put much trust in 3rd party instructions, in particular don't expect you can just copy&paste configs from the internet. Read the `thinkfan.conf` manpage and pay particular attention to the documentation of the `hwmon` path specification and the related `indices` and `name` entries.
